packages/gatsby/cache-dir/page-renderer.js

```diff
@@ -1,35 +1,115 @@
-import React, { createElement } from "react"
+/* global HAS_REACT_18 */
+import React, { createElement, useEffect } from "react"
 import PropTypes from "prop-types"
 import { apiRunner } from "./api-runner-browser"
 import { grabMatchParams } from "./find-path"
+import { StaticQueryContext } from "gatsby"
+import { VALID_NODE_NAMES } from "./head/constants"
+
+// Calls callback in an effect and renders children
+function Caller({ children, callback }) {
+  useEffect(() => {
+    callback()
+  }, [callback])
+
+  return children
+}
+
+// TODO: We do something similar in production-app, we  should use a take this function, put it in some utils file and share
+function render(el, Component) {
```

Comment:
TODO: We do something similar in production-app, we should use a take this function, put it in some utils file and share